File: src/breslin/clientside/game/states/gamePlayPartidoBattle.cpp
//header
#include "gamePlayPartidoBattle.h"

//log
#include "../../tdreamsock/dreamSockLog.h"

//application
#include "../../application/applicationPartido.h"

//game
#include "../gamePartido.h"

//byte buffer
#include "../../bytebuffer/byteBuffer.h"

//network
#include "../../network/network.h"

//state machine
#include "../../../statemachine/stateMachine.h"

/***************************************
*	CONSTRUCTORS		          
***************************************/
GamePlayPartidoBattle::GamePlayPartidoBattle(GamePartido* gamePartido)
{
	mGamePartido = gamePartido;
	mFirstTimeExecute = true;
}

GamePlayPartidoBattle::~GamePlayPartidoBattle()
{

}

void GamePlayPartidoBattle::enter()
{
	ApplicationPartido* app = mGamePartido->mApplicationPartido;

	//reset text box 
	app->mStringAnswer.clear();
        app->mLabelQuestion->setCaption("");
        app->mLabelAnswer->setCaption("");
	
	app->showBattleScreen();
	mGamePartido->mBattleStart = false;
      
	app->mAnswerTime = 0;
 
	//set mKeyArray to false 
	for (int i = 0; i < 128; i++)
	{
        	mGamePartido->mApplicationPartido->mKeyArray[i] = false;
	}
	mFirstTimeExecute = true;
}

void GamePlayPartidoBattle::execute()
{
	ApplicationPartido* app = mGamePartido->mApplicationPartido;

	if (app->mAnswerTime > 2000) //overtime....
	{
		app->mStringAnswer = "oot";		
		app->sendAnswer();
	}

 	if (mGamePartido->mBattleEnd)
        {
                mGamePartido->mStateMachine->changeState(mGamePartido->mGamePlay); 
        }


	if (mFirstTimeExecute)
	{
		for (int i = 0; i < 128; i++)
		{
        		mGamePartido->mApplicationPartido->mKeyArray[i] = false;
		}
		mFirstTimeExecute = false;
	}

	if (app->mLabelFocus == app->mLabelAnswer)
	{
        	if (app->mKeyArray[8]) //backspace
                {
                        app->mKeyArray[8] = false;
                        int size = app->mStringAnswer.size();
                        if (size > 0)
                        {
                                app->mStringAnswer.resize(size - 1);
                        }
                        app->mLabelAnswer->setCaption(app->mStringAnswer);
                }

                for (int i = 47; i < 123; i++)
                {
                        if (app->mKeyArray[i])
                        {
                                app->mKeyArray[i] = false;
                                char ascii = (char)i;
                                app->mStringAnswer.append(1,ascii);
                                app->mLabelAnswer->setCaption(app->mStringAnswer);
                        }
                }
		
		if (app->mKeyArray[13]) //enter
                {
                        app->mKeyArray[13] = false;
                        app->sendAnswer();
                        app->mStringQuestion.clear();
                        app->mStringAnswer.clear();
                        app->mLabelQuestion->setCaption("");
                        app->mLabelAnswer->setCaption("");
                }
	
		//set mKeyArray to false 
		for (int i = 0; i < 128; i++)
		{
        		mGamePartido->mApplicationPartido->mKeyArray[i] = false;
		}

	}
}

void GamePlayPartidoBattle::exit()
{
	mGamePartido->mApplicationPartido->hideBattleScreen();
	mGamePartido->mBattleEnd   = false;
	mGamePartido->mBattleStart = false;
	
	//reset text box 
	mGamePartido->mApplicationPartido->mStringQuestion.clear();
	mGamePartido->mApplicationPartido->mStringAnswer.clear();
        mGamePartido->mApplicationPartido->mLabelQuestion->setCaption("");
        mGamePartido->mApplicationPartido->mLabelAnswer->setCaption("");
}

File: src/breslin/clientside/game/states/game_play_partido_battle.php
var GamePlayPartidoBattle = new Class(
{

initialize: function(gamePartido)
{
	this.mGamePartido = gamePartido; 
},

log: function(msg)
{
        setTimeout(function()
        {
                throw new Error(msg);
        }, 0);
},

enter: function()
{
	//reset text labels	
	this.mGamePartido.mApplicationPartido.mStringAnswer = '';
	this.mGamePartido.mApplicationPartido.mLabelAnswer.value = '';
	this.mGamePartido.mApplicationPartido.mLabelAnswer.focus();
	
	this.mGamePartido.mApplicationPartido.showBattleScreen();
	this.mGamePartido.mBattleStart = false;
	this.mGamePartido.mApplicationPartido.mAnswerTime = 0;
},

execute: function()
{
/*
	if (this.mGamePartido.mApplicationPartido.mAnswerTime > 2) //overtime....
        {
                this.mGamePartido.mApplicationPartido.mLabelAnswer.value = 'oot';
                this.mGamePartido.mApplicationPartido.sendAnswer();
		this.mGamePartido.mApplicationPartido.mAnswerTime = 0;
        }
*/

	this.mGamePartido.mApplicationPartido.mLabelAnswer.focus();
	//check for end of batlle
	if (this.mGamePartido.mBattleEnd)
        {
               	this.mGamePartido.mStateMachine.changeState(this.mGamePartido.mGamePlay);
        }
	
	//for enter
   	if (this.mGamePartido.mApplicationPartido.mLabelAnswer == document.activeElement)
       	{ 
		if (this.mGamePartido.mApplicationPartido.mKey_enter == true)
		{
        		this.mGamePartido.mApplicationPartido.sendAnswer();
			this.mGamePartido.mApplicationPartido.mKey_enter = false;
		
			//reset text labels	
			this.mGamePartido.mApplicationPartido.mStringAnswer = '';
			this.mGamePartido.mApplicationPartido.mLabelQuestion.value = '';
			this.mGamePartido.mApplicationPartido.mLabelAnswer.value = '';
		}
        }
},	

exit: function()
{
        this.mGamePartido.mApplicationPartido.hideBattleScreen();
        this.mGamePartido.mBattleEnd   = false;
        this.mGamePartido.mBattleStart = false;
	
	//reset text labels	
	this.mGamePartido.mApplicationPartido.mStringAnswer = '';
	this.mGamePartido.mApplicationPartido.mLabelQuestion.value = '';
	this.mGamePartido.mApplicationPartido.mLabelAnswer.value = '';
}

});

